file_snapshot.go

```diff
@@ -384,6 +384,11 @@ func (s *FileSnapshotSink) Close() error {
 	// Close the open handles
 	if err := s.finalize(); err != nil {
 		s.logger.Printf("[ERR] snapshot: Failed to finalize snapshot: %v", err)
+		s.logger.Printf("[INFO] snapshot: Deleting temporary files...")
+		if delErr := os.RemoveAll(s.dir); delErr != nil {
```

Comment:
Yes this is a temp dir - this is at the point where its about to save to a temp directory and it fails. So we can delete that temp directory before returning. 
Tested locally, unfortunately we don't use interfaces to wrap file system calls so that I could mock out an error in a unit test. That's a much larger refactor so I didn't do that right now.